internal/storage/pmem/pmem.go

```diff
@@ -44,69 +50,168 @@ func mountInternal(source, target string) (err error) {
 	return nil
 }
 
-// Mount mounts the pmem device at `/dev/pmem<device>` to `target`.
+// Mount mounts the pmem device at `/dev/pmem<device>` to `target` in a basic scenario.
+// In the case when mappingInfo or/and verityInfo is/are not nil, device-mapper framework
+// is used to create linear and verity targets accordingly.
 //
 // `target` will be created. On mount failure the created `target` will be
 // automatically cleaned up.
 //
 // Note: For now the platform only supports readonly pmem that is assumed to be
 // `ext4`.
-func Mount(ctx context.Context, device uint32, target string) (err error) {
+//
+// Note: both mappingInfo and verityInfo can be non-nil at the same time, in that case
+// linear target is created first and it becomes the data/hash device for verity target.
+func Mount(ctx context.Context, device uint32, target string, mappingInfo *prot.DeviceMappingInfo, verityInfo *prot.DeviceVerityInfo) (err error) {
 	_, span := trace.StartSpan(ctx, "pmem::Mount")
 	defer span.End()
 	defer func() { oc.SetSpanStatus(span, err) }()
 
 	span.AddAttributes(
-		trace.Int64Attribute("device", int64(device)),
+		trace.Int64Attribute("deviceNumber", int64(device)),
 		trace.StringAttribute("target", target))
 
-	source := fmt.Sprintf(pMemFmt, device)
-	return mountInternal(source, target)
+	devicePath := fmt.Sprintf(pMemFmt, device)
+	if mappingInfo == nil && verityInfo == nil {
+		return mountInternal(devicePath, target)
+	}
```

Comment:
This is redundant. The function has the same behavior with or without this block.

Reply:
Hopefully it was clear that I'm referring to the if block only. The devicePath assignment is still needed.